specs/CoreWebView2.CreateFromCOMObject.md

```diff
@@ -64,10 +114,48 @@ namespace Microsoft.Web.WebView2.Core
     /// <exception cref="ArgumentNullException">Thrown when the provided COM pointer is null.</exception>
     /// <exception cref="InvalidComObjectException">Thrown when the value is not an ICoreWebView2 COM object and cannot be wrapped.</exception>
     public static CoreWebView2 CreateFromComICoreWebView2(IntPtr value);
+
+    /// <summary>
+    /// Returns the existing COM ICoreWebView2 object underlying this .NET CoreWebView2 object.
+    /// This allows interacting with the WebView2 control using COM APIs,
+    /// even if the control was originally created using .NET.
+    /// </summary>
+    /// <returns>Pointer to a COM object that implements the ICoreWebView2 COM interface.</returns>
+    public IntPtr GetComICoreWebView2();
   }
 }
 ```
 
+## WinRT COM Interop API
+
+```c# (but really COM IDL)
+/// Interop interface for the CoreWebView2 WinRT object to allow WinRT end
+/// developers to be able to use COM interfaces as parameters for some methods.
+/// This interface is implemented by the Microsoft.Web.WebView2.Core.CoreWebView2
+/// runtime class.
+[uuid(B151AD7C-CFB0-4ECF-B9B2-AFCA868581A6), object, pointer_default(unique)]
+interface ICoreWebView2Interop2 : IUnknown {
+  /// Get a COM ICoreWebView2 interface corresponding to this WinRT CoreWebView2
+  /// object.
+  HRESULT GetComICoreWebView2([out, retval] ICoreWebView2** coreWebView2);
+}
+
+/// Interop interface for the CoreWebView2 WinRT activation factory object to allow
+/// WinRT end developers to be able to use COM interfaces as parameters for some
+/// methods.
+/// This interface is implemented by the Microsoft.Web.WebView2.Core.CoreWebView2
+/// activation factory runtime class.
+[uuid(BABBED43-D40E-40CF-B106-8ED65FAE2E7C), object, pointer_default(unique)]
+interface ICoreWebView2ActivationFactoryInterop : IUnknown {
+  /// Creates a CoreWebView2 WinRT object that wraps an existing COM ICoreWebView2 object.
+  /// This allows interacting with the WebView2 control using WinRT,
+  /// even if the control was originally created using COM.
+  HRESULT CreateFromComICoreWebView2([in] ICoreWebView2* coreWebView2Com,
```

Comment:
What happens if you Create twice from the same ICoreWebView2? Do you get two CoreWebView2 objects that both wrap the same ICoreWebView2? So changes to one are visible to the other? (And events are raised on both?)

Is the way to detect this to call GetComICoreWebView2 on both CoreWebView2's and see if you get the same ICoreWebView2 back?

Reply:
Good question. Returning the same CoreWebView2 seems like the better way to go. We can keep a weak ref to the created CoreWebView2 and return that.

Reply:
Returning the same CoreWebView2 contradicts the "create" prefix, which implies that you get a new object each time. Maybe we can live with that.

Reply:
I'm OK with living with that. Or let me know if you have a better name in mind.

Reply:
I like `Get` for that reason and don't see a downside.

You could argue that that obscures the fact that it's a somewhat expensive operation (there's allocations going on). But that's true of many Get() methods, and implying perf is the reason why those methods aren't exposed as a property.

Reply:
I don't have a real preference for either so Get is fine if folks prefer that. 

I named it CreateFromComICoreWebView2 because that's the name we went with for the .NET version of this API. (This PR is an update of an existing API spec doc for the .NET CoreWebView2.CreateFromComICoreWebView2 method that we had previously reviewed and shipped.) In theory I like naming it Create for consistency, but practically its pretty unlikely that someone will be using both the .NET and WinRT versions of this API so I don't think it would cause much confusion and its not a big blocker for me.

Reply:
Matching .Net (`Create`) sounds OK to me

Reply:
Ok I'll leave as Create then. Thanks!